### PR TITLE
Bugfix/#1426 generic exceptions should never be thrown

### PR DESCRIPTION
**Pull request by SergioFQ**
Changed the generic exceptions for specific ones to show the true exceptions

### CODE REVIEW

**Comment by SergioFQ** [comment]
@munterkalmsteiner

File: inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java [hunk 1]
```diff
@@ -72,7 +73,7 @@ public RemoteStringMatchingNerRecommender(Recommender aRecommender)
         featureName = aRecommender.getFeature().getName();
     }
 
-    public void train(String aTrainingRequestJson) throws Exception
+    public void train(String aTrainingRequestJson) throws UIMAException, SAXException, IOException, RecommendationException
```
Comment (munterkalmsteiner) [inline inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java]: This line is very long. Please use checkstyle (see [here](https://inception-project.github.io//releases/0.12.1/docs/developer-guide.html#_setting_up_checkstyle_and_formatting) to identify and fix these kind of issues.

File: inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java [hunk 2]
```diff
@@ -96,8 +97,7 @@ private TrainingRequest deserializeTrainingRequest(String aRequestJson)
         }
     }
 
-    public String predict(String aPredictionRequestJson)
-        throws Exception
+    public String predict(String aPredictionRequestJson) throws IOException, UIMAException, SAXException, RecommendationException
```
Comment (munterkalmsteiner) [inline inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java]: Long line.

**Comment by SergioFQ** [comment]
Updating without long lines @munterkalmsteiner

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by reckart** [comment]
Jenkins complains that FileUploadDownloadHelper.java line 60 is too long. You should get the same result if you build this PR on the command line.

**Comment by SergioFQ** [comment]
Ok! I´ll fix that.

**Comment by SergioFQ** [comment]
I´ve updated with no long lines. @reckart

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by munterkalmsteiner** [comment]
> Thanks :) Looks good. The commit comments do not conform to our conventions. I'll squash-merge it unless somebody wants to re-write the comments :)

@SergioFQ: here are [instructions](https://help.github.com/en/github/committing-changes-to-your-project/changing-a-commit-message) on how to fix commit messages in order to comply to the [code contribution requirements](https://dkpro.github.io/contributing/) (see second to last bullet in "Preparing a pull request").

**Comment by SergioFQ** [comment]
@reckart I saw Jenkins failed again, why did that happen?

**Comment by reckart** [comment]
Checkstyle:

```
19:20:54 [INFO] There are 2 errors reported by Checkstyle 8.20 with inception/checkstyle.xml ruleset.
19:20:54 [ERROR] ../../../../../../srv/home/svc_jenkins/workspace/INCEpTION (GitHub) (PR builder)/git/inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java:[30] (imports) CustomImportOrder: Importe ohne Gruppe sollten am Ende der Import-Liste stehen: 'de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationException'.
19:20:54 [ERROR] ../../../../../../srv/home/svc_jenkins/workspace/INCEpTION (GitHub) (PR builder)/git/inception-imls-external/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/RemoteStringMatchingNerRecommender.java:[31] (imports) CustomImportOrder: 'org.apache.commons.io.IOUtils' sollte von vorheriger Import-Gruppe durch eine leere Zeile abgesetzt sein.
```

(Sorry for the German messages - try building with Maven on the command-line, then you should get the same message in your preferred language).

**Comment by SergioFQ** [comment]
Updated @reckart

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by SergioFQ** [comment]
@reckart 
I used Maven to know why failed again and I saw it failed cause of ImageFeatureEditor class but I made no changes in that file. Do you have any idea about why did that happen? This is the error: 
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project inception-image: Compilation failure: Compilation failure:
[ERROR] .../inception/inception-image/src/main/java/de/tudarmstadt/ukp/inception/image/feature/ImageFeatureEditor.java:[38,8] de.tudarmstadt.ukp.inception.image.feature.ImageFeatureEditor is not abstract and does not override abstract method getFocusComponent() in de.tudarmstadt.ukp.clarin.webanno.api.annotation.feature.editor.FeatureEditor
[ERROR] .../inception/inception-image/src/main/java/de/tudarmstadt/ukp/inception/image/feature/ImageFeatureEditor.java:[90,22] getFocusComponent() in de.tudarmstadt.ukp.inception.image.feature.ImageFeatureEditor cannot override getFocusComponent() in de.tudarmstadt.ukp.clarin.webanno.api.annotation.feature.editor.FeatureEditor
[ERROR]   return type org.apache.wicket.Component is not compatible with org.apache.wicket.markup.html.form.FormComponent
[ERROR] .../inception/inception-image/src/main/java/de/tudarmstadt/ukp/inception/image/feature/ImageFeatureEditor.java:[89,5] method does not override or implement a method from a supertype
```

**Comment by wpmp** [comment]
@SergioFQ this is dues to some changes in WebAnno, inception-project/inception#1491 will fix this.

**Comment by SergioFQ** [comment]
I used Maven again and no error appears. @reckart

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by reckart** [comment]
Jenkins still reports that in `RemoteStringMatchingNerRecommender.java:[54]`
the import `de.tudarmstadt.ukp.inception.recommendation.api.recommender.RecommendationException` should come before the import `de.tudarmstadt.ukp.inception.recommendation.imls.stringmatch.StringMatchingRecommenderTraits`.

**Comment by SergioFQ** [comment]
I had problems setting up checkstyle so that error didn´t appear to me. I made the change in the order of imports, I hope it works now. @reckart

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Comment by reckart** [comment]
I'll squash-merge since the commit comment format does not conform to our conventions.

**Pull request merged**